MsgPack/Formatters/TypeFormatter.cs

```diff
@@ -227,8 +227,18 @@ private static void BuildSerializeArrayBody(Type type, ILGenerator g, MemberInfo
 
 							// Value
 							g.Emit(OpCodes.Ldarg_0);
-							g.Emit(OpCodes.Ldarg_1);
-							g.EmitCall(OpCodes.Call, property.GetGetMethod(), null);
+							// if it's a struct.. we handle it differently
+							// using the address of the argument and not its value.
+							if (type.IsValueType)
+							{
+								g.Emit(OpCodes.Ldarga_S, 1);
+								g.EmitCall(OpCodes.Call, property.GetGetMethod(), null);
+							}
+							else
+							{
+								g.Emit(OpCodes.Ldarg_1);
+								g.EmitCall(OpCodes.Call, property.GetGetMethod(), null);
```

Comment:
maybe move the duplicate 
```g.EmitCall(OpCodes.Call, property.GetGetMethod(), null);``` out of the if. 
the only change seems to be the other Emit.